pkg/ingester/flush.go

```diff
@@ -441,9 +441,11 @@ func (i *Ingester) flushChunks(ctx context.Context, fp model.Fingerprint, labelP
 		)
 
 		// encodeChunk mutates the chunk so we must pass by reference
+		chunkMtx.Lock()
```

Comment:
Same as toWireChunks. encodeChunk ultimately writes to `c.blocks[*].offset` in `pkg/chunkenc/memchunk.go:674`